pkg/ibm_plugin/server/plugin.go

```diff
@@ -531,6 +603,111 @@ func (s *IBMPluginServer) DeletePermitListRules(ctx context.Context, req *paragl
 	return &paragliderpb.DeletePermitListRulesResponse{}, nil
 }
 
+func (s *IBMPluginServer) CreateVpnGateway(ctx context.Context, req *paragliderpb.CreateVpnGatewayRequest) (*paragliderpb.CreateVpnGatewayResponse, error) {
+	rInfo, err := getResourceMeta(req.Deployment.Id)
+	if err != nil {
+		return nil, err
+	}
+	region, err := s.getRegionOfAddressSpace(rInfo.ResourceGroup, req.Deployment.Namespace, req.AddressSpace)
+	if err != nil {
+		return nil, err
+	}
+	if region == ""{
+		return nil, fmt.Errorf("Failed to find a region where the %v is deployed", req.AddressSpace)
+	}
+	cloudClient, err := s.setupCloudClient(rInfo.ResourceGroup, region)
+	if err != nil {
+		return nil, err
+	}
+	ipAddresses, err := cloudClient.CreateRouteBasedVPN(req.Deployment.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	return &paragliderpb.CreateVpnGatewayResponse{GatewayIpAddresses: ipAddresses}, nil
+}
+
+// returns the region of the VPC containing the specified address space 
+func (s *IBMPluginServer) getRegionOfAddressSpace(resourceGroup, namespace, addressSpace string) (string, error) {
+	client, err:= s.setupCloudClient(resourceGroup, defaultRegion)
+	if err != nil {
+		return "", err
+	}
+	vpcsData, err:= client.GetParagliderTaggedResources(sdk.VPC, []string{namespace}, sdk.ResourceQuery{})
+	if err!=nil{
+		return "", err
+	}
+	for _,vpcData:=range vpcsData{
+		client, err:= s.setupCloudClient(resourceGroup, vpcData.Region)
+		if err!=nil{
+			return "", err
+		}
+		VPCFound,err:=client.IsRemoteInVPC(vpcData.ID, addressSpace)
+		if err!=nil{
+			return "", err
+		}	
+		if VPCFound{
+			return vpcData.Region, nil
+		}
+	}
+	return "", nil
+}
```

Comment:
Linting is really off wrt to spacing

Reply:
@seankimkdy I don't see what's off here. `make lint` doesn't raise an error.
Shouldn't `make lint` (and the associated git workflow) have picked up on that?